torch/csrc/inductor/aoti_runtime/model.h

```diff
@@ -77,6 +78,42 @@ AOTI_NOINLINE static void throw_exception(
     throw_exception(#call, __FILE__, __LINE__); \
   }
 
+AOTI_NOINLINE static void __aoti_check(
+    bool cond,
+    const char* func,
+    const char* file,
+    uint32_t line,
+    const char* msg) {
+#if defined(__CUDACC__)
+  if (!cond) {
+#else
+  if (C10_UNLIKELY(!cond)) {
+#endif
+    std::stringstream ss;
+    ss << "AOTI_CHECK failed at line " << line << ", func " << func << ", file "
+       << file << ", msg " << msg;
+    throw std::runtime_error(ss.str());
+  }
+}
+
+#ifdef STRIP_ERROR_MESSAGES
```

Comment:
We need to update codecache.py to take this macro, otherwise users might be surprised by this behavior.

Reply:
> We need to update codecache.py to take this macro, otherwise users might be surprised.

Could you elaborate? Do you mean we need to pass this macro from the compiler invocation command-line? I think this macro is only used for production mobile builds.

Reply:
I think we can implement by implementing `__aoti_check` and `AOTI_CHECK` in C shim, and then naturally that will take whatever macro was used for building PyTorch. Another benefit of that is we can still call the same implementation in c10 (except we wrap with AOTI_NOINLINE) instead of having a potentially diverged behavior.